tests/phpunit/ConnectionTest.php

```diff
@@ -20,89 +17,82 @@
 class ConnectionTest extends TestCase
 {
     /**
-     * @dataProvider falseConnectionProvider
+     * @dataProvider invalidConnectionProvider
      */
     public function testFalseConnection(AdapterInterface $adapter): void
     {
-
         $fs = new Filesystem($adapter);
 
         $extractor = new FtpExtractor(false, $fs, new NullLogger());
         $this->expectException(UserException::class);
-        $extractor->copyFiles('source', 'destination', new FileStateRegistry([]));
-    }
-
-    public function testFalseSftpConnection(): void
-    {
-        $config = new Config(
-            [
-                'parameters' => [
-                    'host' => 'ftp',
-                    'username' => 'ftpuser',
-                    '#password' => 'userpass',
-                    'port' => 21,
-                    'path' => 'abs',
-                    'connectionType' => 'SFTP',
-                    'timeout' => 1,
-                ],
-            ],
-            new ConfigDefinition()
+        $this->expectExceptionMessageRegExp(
+            '/(Could not login)|(getaddrinfo failed)|(Could not connect to)|(Cannot connect to)/'
```

Comment:
Vím, že si vymýšlím, ale čekal jsem:
`public function testFalseConnection($expectedException, $expectedMessageRegExp, AdapterInterface $adapter): void` a že bude tím pádem přiřazené ke každé položce toho data provideru jedna exception, kterou to testuje. A pokud se některý ten test rozbije, tak vidím, co se rozbilo...